crates/shared/src/db_order_conversions.rs

```diff
@@ -123,16 +123,11 @@ pub fn order_class_from(order: &FullOrderDb) -> OrderClass {
         DbOrderClass::Market => OrderClass::Market,
         DbOrderClass::Liquidity => OrderClass::Liquidity,
         DbOrderClass::Limit => OrderClass::Limit(LimitOrderClass {
-            surplus_fee: big_decimal_to_u256(
-                order
-                    .surplus_fee
-                    .as_ref()
-                    .expect("limit orders must have surplus fee set"),
-            )
-            .unwrap(),
-            surplus_fee_timestamp: order
-                .surplus_fee_timestamp
-                .expect("limit orders must have surplus fee timestamp set"),
+            surplus_fee: order
+                .surplus_fee
+                .as_ref()
+                .map(|fee| big_decimal_to_u256(fee).unwrap()),
+            surplus_fee_timestamp: order.surplus_fee_timestamp,
```

Comment:
This is the main change that fixes the panic. This function is not just used for solvable orders so it needs to handle the null case.